app/controllers/confirmations_controller.rb

```diff
@@ -2,6 +2,23 @@ class ConfirmationsController < Devise::ConfirmationsController
   FLASH_MESSAGE = "Email sent! Please contact support at %<email>s if you are "\
                   "having trouble receiving your confirmation instructions.".freeze
 
+  # GET /resource/confirmation?confirmation_token=abcdef
+  def show
+    self.resource = resource_class.confirm_by_token(params[:confirmation_token])
+    yield resource if block_given?
+
+    if resource.errors.empty?
+      set_flash_message!(:notice, :confirmed)
+      if resource.creator?
+        sign_in(resource)
+      else
+        respond_with_navigational(resource) { redirect_to after_confirmation_path_for(resource_name, resource) }
+      end
+    else
+      respond_with_navigational(resource.errors, status: :unprocessable_entity) { render :new }
+    end
+  end
```

Comment:
After a further discussion with @citizen428 I decided that I will override the Devise method instead of calling super. These are the reasons that played a part in making the decisons: 

(a) The code is easier to follow and debug
(b) We do the same thing for `create` action already
(c) We'd be the ones in charge of any code changes to this method, especially since we're changing the behaviour of it. 


@citizen428 when I override it, I get the ffg error `ConfirmationsController#show is missing a template for request formats: text/html`. I added an empty show template for now to bypass the error but I'm very sure that this is **not** the way to go, do you have any ideas on how to get around it?

Reply:
@Ridhwanathe original implementation of this method responds by redirecting users to to either a failed state or a successful state. This controller method doesn't actually own its own view file.

I'm guessing you get getting this error going down line 13 route. Aside from signing, you should then also redirect the user to a certain page, in other word, other view file, which will solve the problem you are seeing. As as example, see https://github.com/forem/forem/blob/f11e7b49d4f10c36be4aa76fd60672f862ebb794/app/controllers/registrations_controller.rb#L31

Reply:
Thanks Mac, this makes a lot of sense, not sure how i overlooked that 🙈